codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -322,17 +359,12 @@ void returnProcessedBytes(int delta) throws Http2Exception {
         }
 
         void consumeBytes(ChannelHandlerContext ctx, int numBytes) throws Http2Exception {
-            if (stream.id() == CONNECTION_STREAM_ID) {
-                throw new UnsupportedOperationException("Returning bytes for the connection window is not supported");
-            }
-            if (numBytes <= 0) {
-                throw new IllegalArgumentException("numBytes must be positive");
-            }
-
             // Return bytes to the connection window
-            FlowState connectionState = connectionState();
-            connectionState.returnProcessedBytes(numBytes);
-            connectionState.writeWindowUpdateIfNeeded(ctx);
+            if (stream.id() != CONNECTION_STREAM_ID) {
```

Comment:
Maybe we should make a static method to do this...instead of having 1 state object reach out and impact another state object?  Then we could just directly call either the static method, or the connection state's consume bytes.

```java
public static void consumeBytesConnectionAndStream(FlowState streamState) {
FlowState connectionState = state(connectionId);
connectionState.doStuff;

streamState.doStuff;
}
```